flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/LeaderIdRegistry.java

```diff
@@ -15,11 +15,27 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package org.apache.flink.runtime.highavailability;
 
-package org.apache.flink.runtime.resourcemanager;
+import java.util.UUID;
 
-import java.io.Serializable;
+/**
+ * Registry class to keep track of the current leader ID.
+ */
+public class LeaderIdRegistry {
```

Comment:
What exactly do you mean? The class is thread-safe and documented (though documentation can be improved). There is no need for locking. Do you mean marking the leaderSessionID `volatile`? It should be fine if leader changes propagate lazily.